pkg/gzip/gzip.go

```diff
@@ -43,132 +32,11 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
-				fpath,
-				dest,
-			)
-		}
-
-		parent := filepath.Dir(fpath)
-		if err := util.MkdirChownAll(parent, ownerUID, ownerGID, 0755); err != nil {
-			return util.Errorf(
-				"error creating directory %s (parent of %s): %s",
-				parent,
-				hdr.Name,
-				err,
-			)
-		}
-
-		switch hdr.Typeflag {
-		case tar.TypeSymlink:
-			err = os.Symlink(hdr.Linkname, fpath)
-			if err != nil {
-				return util.Errorf(
-					"error creating symlink %s -> %s: %s",
-					fpath,
-					hdr.Linkname,
-					err,
-				)
-			}
-			err = os.Lchown(fpath, uid, gid)
-			if err != nil {
-				return util.Errorf("error setting owner of %s: %s", fpath, err)
-			}
-		case tar.TypeLink:
-			// If you include a file multiple times in an invocation of
-			// Gnu tar, it stores anything after the first as a hard link
-			// to itself.  Since such a structure can't otherwise exist, we
-			// can simply skip it.
-			if hdr.Name == hdr.Linkname {
-				continue
-			}
-			// hardlink paths are encoded relative to the tarball root, rather than
-			// the path of the link itself, so we need to resolve that path
-			linkTarget, err := filepath.Rel(filepath.Dir(hdr.Name), hdr.Linkname)
-			if err != nil {
-				return util.Errorf(
-					"error resolving link: %s -> %s: %s",
-					fpath,
-					hdr.Linkname,
-					err,
-				)
-			}
-			// we can't make the hardlink right away because the target might not
-			// exist, so we'll just make a symlink instead
-			err = os.Symlink(linkTarget, fpath)
-			if err != nil {
-				return util.Errorf(
-					"error creating symlink %s -> %s (originally hardlink): %s",
-					fpath,
-					linkTarget,
-					err,
-				)
-			}
-		case tar.TypeDir:
-			err = os.Mkdir(fpath, hdr.FileInfo().Mode())
-			if err != nil && !os.IsExist(err) {
-				return util.Errorf("error creating directory %s: %s", fpath, err)
-			}
-
-			err = os.Chown(fpath, uid, gid)
-			if err != nil {
-				return util.Errorf("error setting ownership of %s: %s", fpath, err)
-			}
-		case tar.TypeReg, tar.TypeRegA:
-			// Extract the file inside a closure to limit the scope of its open FD
-			err = func() (innerErr error) {
-				f, err := os.OpenFile(
-					fpath,
-					os.O_WRONLY|os.O_CREATE|os.O_TRUNC,
-					hdr.FileInfo().Mode(),
-				)
-				if err != nil {
-					return util.Errorf("error creating %s: %s", fpath, err)
-				}
-				// Released at end of "case" statement
-				defer func() {
-					if closeErr := f.Close(); innerErr == nil {
-						innerErr = closeErr
-					}
-				}()
-
-				err = f.Chown(uid, gid)
-				if err != nil {
-					return util.Errorf("error setting file ownership of %s: %s", fpath, err)
-				}
-
-				_, err = io.Copy(f, tr)
-				if err != nil {
-					return util.Errorf("error extracting to %s: %s", fpath, err)
-				}
-				return nil
-			}()
-			if err != nil {
-				return err
-			}
-		default:
-			return util.Errorf("unhandled type flag %q (header %v)", hdr.Typeflag, hdr)
+	cmd := exec.Command("tar", "xzf", filename, "-C", dest)
+	if currentUser.Username != owner {
```

Comment:
I'm sad to say that removing it... causes tests to fail, since we don't run the tests as root, and non-root users understandably cannot specify a user to switch to. Not even themselves.